internal/services/bot/bot_resource.go

```diff
@@ -0,0 +1,313 @@
+package bot
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/botservice/mgmt/2021-03-01/botservice"
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/location"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/bot/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceBot() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceBotCreate,
+		Read:   resourceBotRead,
+		Update: resourceBotUpdate,
+		Delete: resourceBotDelete,
+		Importer: pluginsdk.ImporterValidatingResourceIdThen(func(id string) error {
+			_, err := parse.BotServiceID(id)
+			return err
+		}, func(ctx context.Context, d *pluginsdk.ResourceData, meta interface{}) ([]*pluginsdk.ResourceData, error) {
+			client := meta.(*clients.Client).Bot.BotClient
+
+			id, err := parse.BotServiceID(d.Id())
+			if err != nil {
+				return nil, err
+			}
+
+			resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+			if err != nil {
+				if utils.ResponseWasNotFound(resp.Response) {
+					return nil, fmt.Errorf("Bot %q was not found in Resource Group %q", id.Name, id.ResourceGroup)
+				}
+
+				return nil, fmt.Errorf("retrieving Bot %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
```

Comment:
The parsed id prints as `Bot Service: (Name xyz / Resource Group xyz)`, so these error messages can be shortened to:
```suggestion
					return nil, fmt.Errorf("%s was not found", *id)
				}

				return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
```